content/docs/command-reference/checkout.md

```diff
@@ -33,15 +33,16 @@ for more details.
 
 The execution of `dvc checkout` does the following:
 
-- Scans the `.dvc` and `dvc.lock` files to compare against the data files or
-  directories in the <abbr>workspace</abbr>. DVC knows which data
-  (<abbr>outputs</abbr>) match because the corresponding hash values are saved
-  in the `outs` fields in those files. Scanning is limited to the given
-  `targets` (if any). See also options `--with-deps` and `--recursive` below.
+- Scans all `dvc.lock` and `.dvc` files to compare the hash values of its
+  <abbr>outputs</abbr> against the actual data files or directories in the
+  workspace (similar to `dvc status`). Scanning is limited to any given
+  `targets` (files inside directories
```

Comment:
By default scans all ...
...
`targets` limits which files to checkout. It can accept `.dvc` files, `dvc.yaml` stages, or path to a file or a directory you'd like to checkout (even if file path is inside the directory tracked as a whole).

we also should come up with a better link, I don't think that it's good to point to example to be honest

Reply:
That's definitely better, but kind of out of place (same as my version) since this bullet is supposed to be a step that checkout takes in it's execution. Best to keep it short probably, so I extracted the targets explanation to a new paragraph before all this, instead (in a36ba41f). WDYT?

Reply:
P/s note I changed all dvc.yaml to dvc.lock in this ref.